tests/api/runtimes/test_dask.py

```diff
@@ -512,6 +514,12 @@ def test_enrich_dask_cluster(self):
         assert scheduler_pod.spec.containers[0].resources == expected_resources
         assert worker_pod.spec.containers[0].env == expected_env
         assert scheduler_pod.spec.containers[0].env == expected_env
+        assert (
+            scheduler_pod.spec.node_selector
+            == mlrun.utils.helpers.merge_with_precedence(
+                self.project_default_function_node_selector, function.spec.node_selector
+            )
```

Comment:
it is better practice to validate against an expected value and not a dependency of another function.
for example. if someone removes the value for `self.project_default_function_node_selector` the test still passes though it removes the coverage